### DISCUSSION TITLE
How can I change 'Bearer' prefix in OAuth2PasswordBearer

### DISCUSSION THREAD

**Discussion by DireWereWolf** [fastapi/fastapi#7281] [Questions]
I'm using OAuth2PasswordBearer scheme, the rest of my services has authorisation prefix 'JWT' instead of 'Bearer', how can I change it?

**Comment by behzadmehrabi** [comment]
instead of using `OAuth2PasswordBearer` directly.
create a custom class that inherit from it and override `__call__` method and change `bearer` to `jwt`.
```
class OAuth2PasswordJWT(OAuth2PasswordBearer):
    def __init__(
        self,
        tokenUrl: str,
        scheme_name: Optional[str] = None,
        scopes: Optional[dict] = None,
        auto_error: bool = True,
    ):
        super().__init__(
            tokenUrl=tokenUrl,
            scopes=scopes,
            scheme_name=scheme_name,
            auto_error=auto_error,
        )

    async def __call__(self, request: Request) -> Optional[str]:
        authorization: str = request.headers.get("Authorization")
        scheme, param = get_authorization_scheme_param(authorization)
        if not authorization or scheme.lower() != "jwt":
            if self.auto_error:
                raise HTTPException(
                    status_code=status.HTTP_401_UNAUTHORIZED,
                    detail="Not authenticated",
                    headers={"WWW-Authenticate": "JWT"},
                )
            else:
                return None
        return param
```

**Comment by DireWereWolf** [comment]
Thanks a lot

**Comment by tiangolo** [comment]
Thanks for the help here @Bezmehrabi ! :clap: :bow: 

Thanks for reporting back and closing the issue @DireWereWolf  :+1:

Have in mind that "Bearer" is just the format for a token sent in the header, which is integrated with OAuth2, JWT as a prefix is probably not part of any standard and would probably not have as much compatibility as "Bearer". Also, "Bearer" doesn't imply anything about the token, so it would most commonly be JWT, and it could be something else as well.

**Comment by Shreyansh-Agrawal** [comment]
In fast api for jwt authentication i am using this 

`oauth2_bearer = OAuth2PasswordBearer(tokenUrl='v1/login')`

if my response data format is like this:
```
{
    "access_token": "token",
    "refresh_token": "token"
}
```
then swagger is able to access this token and add it to its authorization header
but if my response is in this format
```
{
    "code": 200,
    "status": "Ok",
    "message": "Successfully Logged In!",
    "data": {
        "access_token": "token",
        "refresh_token": "token",
        "password_type": "permanent"
    }
}
```
Then its not able to access the 'access_token' field. I want it to be able to go in the 'data' field and then look for 'access_token' field.

How can I do that? How should I override the OAuth2PasswordBearer?
